util/git/client.go

```diff
@@ -303,6 +303,13 @@ func (m *nativeGitClient) Checkout(revision string) error {
 			return err
 		}
 	}
+	if _, err := os.Stat(m.root + "/.gitmodules"); !os.IsNotExist(err) {
+		if submoduleEnabled := os.Getenv(common.EnvGitSubmoduleEnabled); submoduleEnabled != "false" {
```

Comment:
don't you want to flip this condition?

Reply:
If I flip it then we have to set a value for it instead of accepting the default is true if it is not set.